web/libs/core/src/lib/hooks/useCurrentUser.ts

```diff
@@ -0,0 +1,61 @@
+import type { APIUser } from "@humansignal/core/types/user";
+import { API } from "apps/labelstudio/src/providers/ApiProvider";
```

Comment:
```
import { API } from "apps/labelstudio/src/providers/ApiProvider";
```

How does this import even work, we are importing ApiProvider from the application into a lib, then using this lib in the application?

Reply:
yeah, kinda like that :D the thing is that `ApiProvider` is where we initialize the APIProxy instance. it has all the necessary configuration and whatnot. while I believe this should also be a part of the core, I decided not to do that in this PR as it brings a lot of changes already and I didn't want to make it bigger

the plan is to have a core version of APIProxy instance that both LSO and LSE can extend on. unfortunately, APIProxy currently doesn't allow extending configuration on an already initialized instance, so that's something we need to do prior to moving API to core

for now unfortunately that's the only way to have access to API in the core lib